### DISCUSSION TITLE
RegressionTask Error

### DISCUSSION THREAD

**Discussion by Seifsherif1111** [torchgeo/torchgeo#2657] [Q&A] [answered]
I get this error whenever I try to fit the trainer with the regression task saying RegressionTask is not a lightingModule nor a torch._dynamo.OptimizedModule
![image](https://github.com/user-attachments/assets/296b4730-a63c-48a7-b55c-ee655bb005cc)

Also I appreciate this package so much but I'd recommend adding a way to make a rasterdataset that you can add ground truth values for it easily csv would be so helpful

**Comment by Seifsherif1111** [comment]
also this is how I defined the task
```py 

task = RegressionTask(model = 'resnet50',backbone='resnet50',weights="ResNet50_Weights.SENTINEL2_ALL_MOCO",in_channels=9,num_outputs=11,loss='mse',lr=0.0001,patience = 10,freeze_backbone=True)
```

**Comment by isaaccorley** [comment, marked as answer]
Can you share the rest of your notebook? This is likely because you are importing `pytorch_lightning.Trainer` when it should be `lightning.Trainer`

**Reply by Seifsherif1111** [reply]
here's the link for the notebook 
https://colab.research.google.com/drive/1DpBm02lHqt-AmIXNO6A5hU2bM6ZeiweY?usp=sharing
I am really importing pytorch_lighting but the lightning trainer doesn't work since I created my module based on the pytorch_lightning module, would changing it to lighting module work?

**Reply by isaaccorley** [reply]
Yes you will need to upgrade to lightning instead of pytorch-lightning because that is what our modules use.

**Comment by adamjstewart** [comment]
> I'd recommend adding a way to make a rasterdataset that you can add ground truth values for it easily csv would be so helpful

Can you tell me more about this use case? What would the CSV file look like? I've never seen a dataset in this format before, so it doesn't sound like a common use case.

**Reply by Seifsherif1111** [reply]
Yes I realized that this use case isn't common at all. but for my case I'm trying to train a model to predict numerical values from satellite imagery, so I have the raster images of the samples and the numeric properties of the samples as labels. I had to implement this module that supports this use case from scratch so I thought it'd be helpful to encourage similar tasks to arise if this data module was already implemented and ready for use.

**Reply by adamjstewart** [reply]
I definitely want to encourage more regression datasets, but without existing examples it's difficult to design something generic enough to handle them all. Perhaps we can wait for a few more datasets to be added to TorchGeo.
